src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -209,13 +204,26 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     request.Headers.TransferEncodingChunked = true;
                 }
 
+                if (request.Version.Major != 1)
+                {
+                    // TODO #23134: Support 2.0
+                    throw new PlatformNotSupportedException($"Only HTTP 1.0 supported -- request.Version was {request.Version}");
+                }
+                else if (request.Version.Minor == 0 && request.Headers.TransferEncodingChunked == true)
+                {
+                    // HTTP 1.0 does not support chunking
+                    throw new NotSupportedException($"HTTP 1.0 does not support chunking.");
+                }
+
                 // Write request line
                 await WriteStringAsync(request.Method.Method, cancellationToken).ConfigureAwait(false);
                 await WriteByteAsync((byte)' ', cancellationToken).ConfigureAwait(false);
                 await WriteStringAsync(
                     _usingProxy ? request.RequestUri.AbsoluteUri : request.RequestUri.PathAndQuery,
                     cancellationToken).ConfigureAwait(false);
-                await WriteBytesAsync(s_spaceHttp11NewlineAsciiBytes, cancellationToken).ConfigureAwait(false);
+                await WriteBytesAsync(
+                    request.Version.Minor == 1 ? s_spaceHttp11NewlineAsciiBytes : s_spaceHttp10NewlineAsciiBytes,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
I think there are some other behaviors that would need to be tweaked as well.  For example, are we supposed to respect Expect: 100-continue in 1.0?  Isn't there a difference between 1.1 and 1.0 about connections remaining open vs closed by default?

Reply:
yes, I was wondering about that. As I mentioned in notes, this may not be complete fix. 
I read through some compatibility notes and in some cases the best behavior is not clear. 
As long as we send valid 1.0 request server should not respond with 1.1 features. 
If it ever does, we may either ignore them, throw exception or process them as 1.1.

Reply:
>Isn't there a difference between 1.1 and 1.0 about connections remaining open vs closed by default?

Short answer is yes there is a difference in defaults about "Connection: close".  Persistent connections in general were an HTTP/1.1 feature.

Reply:
I did set up old 1.3 Apache (from 1998) 

```
test@dale:~$ curl -v --http1.0 http://127.0.0.1:8013/ -H 'User-Agent:  MSIE 4.0b2;'
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8013 (#0)
> GET / HTTP/1.0
> Host: 127.0.0.1:8013
> Accept: */*
> User-Agent:  MSIE 4.0b2;
>
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Date: Wed, 30 Aug 2017 04:20:38 GMT
< Server: Apache/1.3.0 (Unix)
< Last-Modified: Wed, 20 May 1998 14:59:42 GMT
< ETag: "3c31e8-656-3562efde"
< Accept-Ranges: bytes
< Content-Length: 1622
< Connection: close
< Content-Type: text/html
<
```
also without the  forced IE workaround server would respond with HTTP/1.1 response to HTTP/1.0 request with Connection: close header 

```
test@dale:~$ curl -v --http1.0 http://127.0.0.1:8013/
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8013 (#0)
> GET / HTTP/1.0
> Host: 127.0.0.1:8013
> User-Agent: curl/7.50.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Wed, 30 Aug 2017 04:22:32 GMT
< Server: Apache/1.3.0 (Unix)
< Last-Modified: Wed, 20 May 1998 14:59:42 GMT
< ETag: "3c31e8-656-3562efde"
< Accept-Ranges: bytes
< Content-Length: 1622
< Connection: close
< Content-Type: text/html
```
I'll do more testing tomorrow.

Reply:
I did little bot more testing with various servers. 
We may get 1.0 or 1.1 response to 1.0 request. 
The observer behavior may change based on User-Agent. 
Some servers have fixup for MSIE 4.0 (and few other browsers)

Connection: close may or may not be set.
In all cases servers did send FIN after body.
The 100 code was introduced in 1.1 and it should be used for 1.0 clients. 
I don't think we need to do anything special to prevent that.

Content-length may or may not be set. 
I tested all scenarios above with managed Handler and I did not observe any strange behavior.(I have simple get app and I dump headers and be body length) 

I also tested what happens when version 1.2 used and we fall back to 1.1 both for current core and for managed handler.

Reply:
> Connection: close may or may not be set. In all cases servers did send FIN after body.

Here we configure the response stream to be used:
https://github.com/dotnet/corefx/blob/1504621ef6455bad4a7c66296dd0e45982793968/src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs#L357-L384
If 1.0 is used, should we always use a response stream that'll close the connection when it completes?

Reply:
>If 1.0 is used, should we always use a response stream that'll close the connection when it completes?

You should check the RFC.   While HTTP/1.0 has by default connections that are not persistent, I believe that it still can support "Connection: Keep-Alive" semantics if explicitly requested.

Reply:
Ah, ok.  Then this is fine.

Reply:
Should we advertise it by default? It seems like this is something useful if BOTH sides are willing to do that.

Reply:
> Should we advertise it by default?

What does that mean?

Reply:
Should we include "Connection: Keep-ALive" in default 1.0 request to indicate we are willing to reuse connection? We do not need to do that for 1.1 as it is default, right?

Reply:
I recommend that the RFCs be consulted.

The semantics for how the "Connection: Keep-Alive" header is used depends on whether it is request or response header.

Reply:
right. The managed handler is client so this would be always request side - at least for now. 
I could not find anything in HTTP 1.0 RFC 1945. There is only ietf draft out there for 1.0 as far as I can tell. 

That suggest that we should keep it out by default. Also there were some notes about interoperability with proxies and current implementation does not do it. 

We can probably assume that if anybody asks explicitly for 1.0, it is not done for performance reasons.